### DISCUSSION TITLE
Model conversion tool

### DISCUSSION THREAD

**Discussion by clizbe** [TulipaEnergy/TulipaEnergyModel.jl#429] [Ideas]
We gain a lot of efficiency with the flexible connections in our graph. But this relies on the user knowing how this works and not modelling in the old way.
Maybe a nice-to-have (as Tars suggested at the eScience Workshop) would be a tool that does this conversion for the user, so it's more fool-proof. This would also probably help for interop with other ESDL-compatible models.

**Comment by abelsiqueira** [comment]
From what I saw in the presentation, maybe this could be achieved by doing something like [edge contraction](https://en.m.wikipedia.org/wiki/Edge_contraction#:~:text=In%20graph%20theory%2C%20an%20edge,the%20theory%20of%20graph%20minors.). I don't know enough about the technique or the problem to know for sure.

**Comment by clizbe** [comment]
Cool there might be a method for it!

I think first we should debate whether we WANT this - because it really lets the user stay "dumb" and not understand how the model works - and there's the risk that the converter makes a mistake and the user has no idea what happened. (Plus even if we want it, it's a bonus feature.)

If there isn't a solid automation to do it consistently, we might just need a section in big red letters in the docs. ;)
Because otherwise there will probably be noobs who don't do it properly and suffer in their performance.

**Comment by clizbe** [comment]
I don't think this is still relevant, since @suvayu is implementing a converter from a fairly "normal" ESDL map file.
